Doc: setting up (#1731)
vtex/faststore: apps/site/pages/docs/getting-started/settingup-project.mdx

</Callout>

## Step 2: Running a local server

Let's check what the project looks like in a web browser so far.

1. Start a local development server to serve your website:

```bash
  yarn dev
```
This may take a few minutes. Once your development server is ready, the command line will output a similar message as the following:

```bash
  ready - started server on 0.0.0.0:3000, url: http://localhost:3000
  event - compiled client and server successfully in 333 ms (951 modules)
```
<Callout type="info" emoji="ℹ️">
  The `dev` command is a part of the FastStore CLI, a pre-configured command line interface in your starter project. This command enables you to execute code to keep your store's project up-to-date with the [@faststore/core package](/docs/getting-started/project-structure) using the CLI. For more information, you can refer to the topic on [FastStore CLI](/docs/getting-started/faststore-cli).
</Callout>

2. Now, open your favorite web browser and navigate to the localhost served, for example, http://localhost:3000.

![localhost-example](https://vtexhelp.vtexassets.com/assets/docs/src/starter-gif___42003e72ece61064991a34c3f14e398f.gif)

> *Congratulations!* You have just created your very first FastStore store website! 🎉

Reviewer: I might have missed it, but is it a good idea to tell people where this content is coming from?

Author: @gvc , do you mean saying that this content comes from the `@faststore/core` package, for example?

While your local development server is running, you can preview any changes you make to your files, and saved changes will hot reload in the browser. To stop the development server, return to the terminal and press `Ctrl + C`. To restart it, run `yarn dev` again.

## Step 3: Making the first change in your store frontend

> Now that your store is up and running locally, let's make a simple change in the store frontend

Reviewer 2: ```suggestion
Now that your store is up and running locally, let's make a simple change in the store frontend.
```

  // --------------------------------------------------------
 
  // FACE
  --fs-text-face-body: 'Lato', -apple-system, system-ui, BlinkMacSystemFont, sans-serif;
 
  // --------------------------------------------------------
  // Refinements
  // --------------------------------------------------------
 
  // BORDERS
  --fs-border-radius: 0.25rem;
 
  // SHADOW
  --fs-shadow: none;
  --fs-shadow-darker: 0 0 10px rgb(0 0 0 / 20%);
  --fs-shadow-hover: 0 1px 4px rgb(0 0 0 / 10%), 0 6px 8px rgb(0 0 0 / 10%);
 
  // --------------------------------------------------------
  // Components
  // --------------------------------------------------------
 
  [data-fs-product-card] {
    --fs-product-card-border-color: transparent;
    --fs-product-card-border-color-hover: var(--fs-border-color-light);
 
    &[data-fs-product-card-bordered='true'] {
      --fs-product-card-border-color: var(--fs-border-color-light);
    }
  }
 
  [data-fs-logo] {
    --fs-logo-img: url('~/public/brandless-positive.png');
    --fs-logo-width: 8rem;
  }
}

```

4. Save your changes.
5. Restart the server and check your browser to see the changes you have made. You may need to refresh the page.

> 

Reviewer 2: Add the preview screenshot of the page with soft-blue theme here.

Reviewer 2: maybe?
<img width="1440" alt="image" src="https://user-images.githubusercontent.com/3356699/235955112-3365d65e-de89-4cf2-bb4c-a0c8322ce387.png">

### Customizing a Component (Optional)
If you are up for another challenge, let's customize a component. We will use the `Listing` variant from the [Price Component](/components/atoms/price) in this tutorial.

> 6. In the `soft-blue.scss` file, add the `[data-fs-price]` and its data attribute `data-fs-price-variant` with the `Listing` value: 

Author: ```suggestion
6. In the `soft-blue.scss` file, add the `data-fs-price` and  `data-fs-price-variant` data attributes with the `Listing` value: 
```

</Callout>

2. Change into the working directory (e.g., cd `faststore.store`) and install all the dependencies listed within the `package.json` file:

```bash
cd faststore.store
yarn install
```

3.  Open up your FastStore project in any code editor of your choice.

<Callout type="info" emoji="ℹ️">
  Note that your store configuration was already set during the [Onboarding](/docs/getting-started/onboarding#repository-and-live-store-created). Therefore, there is no need to connect your project to your VTEX account. 
  If you want to review these configurations, you can access the `faststore.config.js` file in the root directory of your project's repository.
</Callout>

## Step 2: Running a local server

Let's check what the project looks like in a web browser so far.

1. Start a local development server to serve your website:

```bash
  yarn dev
```
This may take a few minutes. Once your development server is ready, the command line will output a similar message as the following:

```bash
  ready - started server on 0.0.0.0:3000, url: http://localhost:3000
  event - compiled client and server successfully in 333 ms (951 modules)
```
<Callout type="info" emoji="ℹ️">

>   The yarn dev command is a part of the FastStore CLI, a pre-configured binary in your starter project. This command enables you to execute code to keep your store's project up-to-date with the [core package](/tbd) using the command-line interface. For more information, you can refer to the topic on [FastStore CLI](/tbd).

Author: ```suggestion
  The `dev` command is a part of the FastStore CLI, a pre-configured command line interface in your starter project. This command enables you to execute code to keep your store's project up-to-date with the [@faststore/core package](/docs/getting-started/project-structure) using the CLI. For more information, you can refer to the topic on [FastStore CLI](/docs/getting-started/faststore-cli).
```

 
  // --------------------------------------------------------
  // Typography (Branding Core)
  // --------------------------------------------------------
 
  // FACE
  --fs-text-face-body: 'Lato', -apple-system, system-ui, BlinkMacSystemFont, sans-serif;
 
  // --------------------------------------------------------
  // Refinements
  // --------------------------------------------------------
 
  // BORDERS
  --fs-border-radius: 0.25rem;
 
  // SHADOW
  --fs-shadow: none;
  --fs-shadow-darker: 0 0 10px rgb(0 0 0 / 20%);
  --fs-shadow-hover: 0 1px 4px rgb(0 0 0 / 10%), 0 6px 8px rgb(0 0 0 / 10%);
 
  // --------------------------------------------------------
  // Components
  // --------------------------------------------------------
 
  [data-fs-product-card] {
    --fs-product-card-border-color: transparent;
    --fs-product-card-border-color-hover: var(--fs-border-color-light);
 
    &[data-fs-product-card-bordered='true'] {
      --fs-product-card-border-color: var(--fs-border-color-light);
    }
  }
 
  [data-fs-logo] {
    --fs-logo-img: url('~/public/brandless-positive.png');
    --fs-logo-width: 8rem;
  }
}

```

> 3. Add the [Lato](https://fonts.google.com/specimen/Lato?query=lato) font on `src/fonts/WebFonts.tsx`:

Author: Delete this step.

```tsx
    function WebFonts() {
    return (
      <link
        rel="stylesheet"
        href="https://fonts.googleapis.com/css2?family=Lato:wght@300;400;700;900&display=swap"
      />
    )
  }
```

> 4. Restart the server.

Author: Switch to step 5

5. Save your changes and check your browser to see the changes you have made. You may need to refresh the page.

> Now, let's customize a component. We will use the `Listing` variant from the [Price Component](/components/atoms/price) in this tutorial.

Author: Make this an optional step

6. In the `soft-blue.scss` file, add the `data-fs-price-variant` data attribute with the `Listing` value: 

```scss
  ...
  [data-fs-logo] {
    --fs-logo-img: url('~/public/brandless-positive.png');
    --fs-logo-width: 8rem;
  }
  

>   [data-fs-price-variant="Listing"]: {

Reviewer 2: ```suggestion
  [data-fs-price-variant="Listing"] {
```

>   [data-fs-price-variant="Listing"] {
>   }

Reviewer 2: ```suggestion
  [data-fs-price] [data-fs-price-variant="listing"] {
  }
```

  
```

7. Now, change the color of the  `Listing` variant using its [local token](/components/atoms/price#design-tokens). For this tutorial, we will use the shade of red `#cb4242`:
```scss
  }
  

>   [data-fs-price-variant="Listing"]: {

Reviewer 2: ```suggestion
  [data-fs-price-variant="Listing"] {
```

>   [data-fs-price-variant="Listing"] {
>     --fs-price-listing-color: #cb4242;
>   }

Reviewer 2: ```suggestion
[data-fs-price][data-fs-price-variant="listing"] {
    --fs-price-listing-color: #cb4242;
  }
```

>     --fs-price-listing-color: #cb4242
>   }
>   

Reviewer 2: ```suggestion
    --fs-price-listing-color: #cb4242;
  }
  
```
